AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -371,30 +371,58 @@ - (void)flushQueue {
 
 - (void)checkPendingLogs {
 
-  // Skip checking pending logs if the channel is paused.
-  if (self.paused) {
-    return;
-  }
-
   // If the interval is default and we reached batchSizeLimit flush logs now.
-  if (self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
+  if (!self.paused && self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
```

Comment:
Looks like `- (void)checkPendingLogs` can be simplified by adding `if (!self.paused)` check upfront:

```
- (void)checkPendingLogs {
  if (!self.paused) {
    
    // If the interval is default and we reached batchSizeLimit flush logs now.
    if (self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
      [self flushQueue];
    } else if (self.itemsCount > 0) {
      NSUInteger flushInterval = [self resolveFlushInterval];
      
      // If the interval is over, send all logs without any additional timers.
      if (flushInterval == 0) {
        [self flushQueue];
      }
      
      // Postpone sending logs.
      else {
        [self startTimer:flushInterval];
      }
    }
  }
}
```